united.py

```diff
@@ -28,10 +54,224 @@ def parse_date(created):
     """
     time = datetime.strptime(created[:-3], '%m/%d/%Y %I:%M:%S')
     hour = time.strftime('%H')
-    hour = int(hour) if created.endswith('AM') else int(hour)+12
+    hour = int(hour) if created.endswith('AM') else int(hour) + 12
     return datetime(time.year, time.month, time.day, hour, time.minute, 0)
 
 
+def is_nth_weekday(nth, daynum, year, month):  # start counting the daynum from monday = 0
+    return calendar.Calendar(nth).monthdatescalendar(
+        year,
+        month
+    )[daynum][0]
+
+
+def get_parent_object_node(node):
+    while node.parentNode:
+        node = node.parentNode
+        if node.nodeName == "Object":
+            return node
+
+
+def all_station_in_date_frame(created):
+
+    station_data_in_date = collection.getElementsByTagName('date_selected')
+    station_data_in_date.sort(key=lambda x: int(x.attributes['Position'].value))
+    accident_date = parse_date(created)
+    summer_clock = is_nth_weekday(4, 4, accident_date[0], 3)
+    winter_clock = is_nth_weekday(4, 6, accident_date[0], 10)
+
+    # weather is given in UTC time
+    # therefore in summer clock we deduct 3 hours from the local time and in winter clock 2 hours
```

Comment:
Or even better, define a constant `TIME_ZONE=2` which you always subtract, and then subtract another 1 only if necessary, and you won't need the `else`.
Also, you can combine the three first conditions with `and` rather than repeating the `if` contents.
Why are you referring to `accident_date` sometimes by attributes and sometimes with brackets?